uniswap/uniswap.py

```diff
@@ -652,6 +655,7 @@ def _token_to_token_swap_input(
         fee: int,
         slippage: float,
         fee_on_transfer: bool = False,
+        route: Optional[List[AddressLike]] = None,
```

Comment:
Same concern as above, needs to be respected in all cases, or raise an exception if it can't.